### ISSUE TITLE
Missing exports

### ISSUE THREAD

**Issue by Alexpux** [mdadams/jasper#122]
Current jasper missing some exports needed by other projects that was in old jasper 1.900.x

````diff -Naur jasper-version-2.0.10-orig/src/libjasper/include/jasper/jas_cm.h jasper-version-2.0.10/src/libjasper/include/jasper/jas_cm.h
--- jasper-version-2.0.10-orig/src/libjasper/include/jasper/jas_cm.h	2016-12-30 19:40:24.000000000 +0300
+++ jasper-version-2.0.10/src/libjasper/include/jasper/jas_cm.h	2017-02-23 23:35:58.961568800 +0300
@@ -256,7 +256,7 @@
 JAS_DLLEXPORT void jas_cmprof_destroy(jas_cmprof_t *prof);
 
 int jas_clrspc_numchans(int clrspc);
-jas_iccprof_t *jas_iccprof_createfromcmprof(jas_cmprof_t *prof);
+JAS_DLLEXPORT jas_iccprof_t *jas_iccprof_createfromcmprof(jas_cmprof_t *prof);
 
 #define	jas_cmprof_clrspc(prof) ((prof)->clrspc)
 JAS_DLLEXPORT jas_cmprof_t *jas_cmprof_copy(jas_cmprof_t *prof);
diff -Naur jasper-version-2.0.10-orig/src/libjasper/include/jasper/jas_icc.h jasper-version-2.0.10/src/libjasper/include/jasper/jas_icc.h
--- jasper-version-2.0.10-orig/src/libjasper/include/jasper/jas_icc.h	2016-12-30 19:40:24.000000000 +0300
+++ jasper-version-2.0.10/src/libjasper/include/jasper/jas_icc.h	2017-02-23 23:35:41.817588200 +0300
@@ -375,32 +375,32 @@
 	jas_iccattrvalops_t ops;
 } jas_iccattrvalinfo_t;
 
-jas_iccprof_t *jas_iccprof_load(jas_stream_t *in);
-int jas_iccprof_save(jas_iccprof_t *prof, jas_stream_t *out);
-void jas_iccprof_destroy(jas_iccprof_t *prof);
-jas_iccattrval_t *jas_iccprof_getattr(jas_iccprof_t *prof,
+JAS_DLLEXPORT jas_iccprof_t *jas_iccprof_load(jas_stream_t *in);
+JAS_DLLEXPORT int jas_iccprof_save(jas_iccprof_t *prof, jas_stream_t *out);
+JAS_DLLEXPORT void jas_iccprof_destroy(jas_iccprof_t *prof);
+JAS_DLLEXPORT jas_iccattrval_t *jas_iccprof_getattr(jas_iccprof_t *prof,
   jas_iccattrname_t name);
-int jas_iccprof_setattr(jas_iccprof_t *prof, jas_iccattrname_t name,
+JAS_DLLEXPORT int jas_iccprof_setattr(jas_iccprof_t *prof, jas_iccattrname_t name,
   jas_iccattrval_t *val);
-void jas_iccprof_dump(jas_iccprof_t *prof, FILE *out);
-jas_iccprof_t *jas_iccprof_copy(jas_iccprof_t *prof);
-int jas_iccprof_gethdr(jas_iccprof_t *prof, jas_icchdr_t *hdr);
-int jas_iccprof_sethdr(jas_iccprof_t *prof, jas_icchdr_t *hdr);
+JAS_DLLEXPORT void jas_iccprof_dump(jas_iccprof_t *prof, FILE *out);
+JAS_DLLEXPORT jas_iccprof_t *jas_iccprof_copy(jas_iccprof_t *prof);
+JAS_DLLEXPORT int jas_iccprof_gethdr(jas_iccprof_t *prof, jas_icchdr_t *hdr);
+JAS_DLLEXPORT int jas_iccprof_sethdr(jas_iccprof_t *prof, jas_icchdr_t *hdr);
 
-void jas_iccattrval_destroy(jas_iccattrval_t *attrval);
-void jas_iccattrval_dump(jas_iccattrval_t *attrval, FILE *out);
-int jas_iccattrval_allowmodify(jas_iccattrval_t **attrval);
-jas_iccattrval_t *jas_iccattrval_clone(jas_iccattrval_t *attrval);
-jas_iccattrval_t *jas_iccattrval_create(jas_iccuint32_t type);
+JAS_DLLEXPORT void jas_iccattrval_destroy(jas_iccattrval_t *attrval);
+JAS_DLLEXPORT void jas_iccattrval_dump(jas_iccattrval_t *attrval, FILE *out);
+JAS_DLLEXPORT int jas_iccattrval_allowmodify(jas_iccattrval_t **attrval);
+JAS_DLLEXPORT jas_iccattrval_t *jas_iccattrval_clone(jas_iccattrval_t *attrval);
+JAS_DLLEXPORT jas_iccattrval_t *jas_iccattrval_create(jas_iccuint32_t type);
 
-void jas_iccattrtab_dump(jas_iccattrtab_t *attrtab, FILE *out);
+JAS_DLLEXPORT void jas_iccattrtab_dump(jas_iccattrtab_t *attrtab, FILE *out);
 
 extern jas_uchar jas_iccprofdata_srgb[];
 extern int jas_iccprofdata_srgblen;
 extern jas_uchar jas_iccprofdata_sgray[];
 extern int jas_iccprofdata_sgraylen;
-jas_iccprof_t *jas_iccprof_createfrombuf(jas_uchar *buf, int len);
-jas_iccprof_t *jas_iccprof_createfromclrspc(int clrspc);
+JAS_DLLEXPORT jas_iccprof_t *jas_iccprof_createfrombuf(jas_uchar *buf, int len);
+JAS_DLLEXPORT jas_iccprof_t *jas_iccprof_createfromclrspc(int clrspc);
 
 #ifdef __cplusplus
 }
diff -Naur jasper-version-2.0.10-orig/src/libjasper/include/jasper/jas_image.h jasper-version-2.0.10/src/libjasper/include/jasper/jas_image.h
--- jasper-version-2.0.10-orig/src/libjasper/include/jasper/jas_image.h	2016-12-30 19:40:24.000000000 +0300
+++ jasper-version-2.0.10/src/libjasper/include/jasper/jas_image.h	2017-02-24 12:54:48.266026300 +0300
@@ -311,7 +311,7 @@
   jas_image_cmptparm_t *cmptparms, jas_clrspc_t clrspc);
 
 /* Create an "empty" image. */
-jas_image_t *jas_image_create0(void);
+JAS_DLLEXPORT jas_image_t *jas_image_create0(void);
 
 /* Clone an image. */
 JAS_DLLEXPORT jas_image_t *jas_image_copy(jas_image_t *image);
diff -Naur jasper-version-2.0.10-orig/src/libjasper/include/jasper/jas_malloc.h jasper-version-2.0.10/src/libjasper/include/jasper/jas_malloc.h
--- jasper-version-2.0.10-orig/src/libjasper/include/jasper/jas_malloc.h	2016-12-30 19:40:24.000000000 +0300
+++ jasper-version-2.0.10/src/libjasper/include/jasper/jas_malloc.h	2017-02-22 19:18:12.162859500 +0300
@@ -89,25 +89,25 @@
 \******************************************************************************/
 
 /* Allocate memory. */
-void *jas_malloc(size_t size);
+JAS_DLLEXPORT void *jas_malloc(size_t size);
 
 /* Free memory. */
-void jas_free(void *ptr);
+JAS_DLLEXPORT void jas_free(void *ptr);
 
 /* Resize a block of allocated memory. */
-void *jas_realloc(void *ptr, size_t size);
+JAS_DLLEXPORT void *jas_realloc(void *ptr, size_t size);
 
 /* Allocate a block of memory and initialize the contents to zero. */
-void *jas_calloc(size_t num_elements, size_t element_size);
+JAS_DLLEXPORT void *jas_calloc(size_t num_elements, size_t element_size);
 
 /* Allocate array (with overflow checking) . */
-void *jas_alloc2(size_t num_elements, size_t element_size);
+JAS_DLLEXPORT void *jas_alloc2(size_t num_elements, size_t element_size);
 
 /* Allocate array of arrays (with overflow checking) . */
-void *jas_alloc3(size_t num_arrays, size_t array_size, size_t element_size);
+JAS_DLLEXPORT void *jas_alloc3(size_t num_arrays, size_t array_size, size_t element_size);
 
 /* Resize a block of allocated memory (with overflow checking) . */
-void *jas_realloc2(void *ptr, size_t num_elements, size_t element_size);
+JAS_DLLEXPORT void *jas_realloc2(void *ptr, size_t num_elements, size_t element_size);
 
 #if defined(JAS_DEFAULT_MAX_MEM_USAGE)
 
````

**Comment by Partha1b** [comment]
Hey, thanks for providing this patch. I was having difficulty building GIMP plugin for jp2-loading and this patch solved that issue. 

Hope it's add to the source.

**Comment by mdadams** [comment]
Thanks for the above patches.  They have been included in commit a0ab51fcfa0d933e909b39866bed3ce2b77255b2.
